deploy/vagrant/conf/alluxio.yml

```diff
@@ -4,13 +4,16 @@
 # {Github | Local | Release}
 Type: Local
 
+# If the version is less than 1.1.0, ie. 1.0.1. This should be false for 1.1.0-SNAPSHOT. {true|false}
+Version_LessThan_1.1.0: false
```

Comment:
If your concern is having two possibly conflicting sources of truth, I suggest parsing the version field in the Ruby script which can be applied to the Spark case as well. Then, for the deploy module the version fields are the source of truth for which version is deployed.
I think putting an implicit dependency on the version script which would only be revealed as a runtime error isn't as clean and would require a similar amount of effort to implement.